types/lodash/fp/index.d.ts

```diff
@@ -0,0 +1,618 @@
+         *
+         * @param func The function to have its output memoized.
+         * @return Returns the new memoizing function.
+         */
+        memoize(func: Function);
+
+        /**
+         * Creates an object composed of keys generated from the results of running each element of collection through
+         * iteratee. The corresponding value of each key is the last element responsible for generating the key. The
+         * iteratee function is bound to thisArg and invoked with three arguments:
+         * (value, index|key, collection).
+         *
+         * If a property name is provided for iteratee the created _.property style callback returns the property
+         * value of the given element.
+         *
+         * If a value is also provided for thisArg the created _.matchesProperty style callback returns true for
+         * elements that have a matching property value, else false.
+         *
+         * If an object is provided for iteratee the created _.matches style callback returns true for elements that
+         * have the properties of the given object, else false.
+         *
+         * @param term
+         * @param collection The collection to iterate over.
+         * @return Returns the composed aggregate object.
+         */
+        keyBy<T>(termn: string): (collection: List<T>) => T;
+        keyBy<T>(termn: string, collection: List<T>): T;
+
+        /**
+         * Gets the first element of array.
+         *
+         * @alias _.first
+         *
+         * @param array The array to query.
+         * @return Returns the first element of array.
+         */
+        head<T>(array: List<T>): T;
+
+        flatten();
+
+        /**
+         * Iterates over elements of collection, returning the first element predicate returns truthy for.
+         * The predicate is bound to thisArg and invoked with three arguments: (value, index|key, collection).
+         *
+         * If a property name is provided for predicate the created _.property style callback returns the property
+         * value of the given element.
+         *
+         * If a value is also provided for thisArg the created _.matchesProperty style callback returns true for
+         * elements that have a matching property value, else false.
+         *
+         * If an object is provided for predicate the created _.matches style callback returns true for elements that
+         * have the properties of the given object, else false.
+         *
+         * @param predicate The function invoked per iteration.
+         * @param collection The collection to search.
+         * @return Returns the matched element, else undefined.
+         */
+        find<T>(predicate: ListIterator<T, boolean>, collection: List<T>): T;
+        find<T>(predicate: ListIterator<T, boolean>): (collection: List<T>) => T;
+
+        /**
+         * @see fp.find
+         */
+        find<TObject extends {}, T>(predicate: TObject, collection: List<T>): T;
```

Comment:
`TObject` is only used once, so it's not a very good type constraint. I think this should be more like `find<T>(predicate: object, collection: List<T>): T;`. This appears to be a common issue in many other function definitions.